packages/docs/stories/components/range.stories.ts

```diff
@@ -0,0 +1,453 @@
+          min-width: 0;
+        }
+      }
+    </style>
+    <script>
+      [...document.querySelectorAll('.suffix-input-field')]
+        .forEach(range => {
+          const input = range.querySelector('syn-input');
+
+          range.addEventListener('syn-input', e => {
+            const { target } = e;
+            if (target.tagName !== 'SYN-RANGE') return;
+            target.querySelector('syn-input').value = target.value;
+          });
+
+          input.addEventListener('syn-input', e => {
+            const { target } = e;
+            target.closest('syn-range').value = target.value;
+          });
+        });
+    </script>
+  `,
+};
+
+export const CustomTrackColors: Story = {
+  parameters: {
+    docs: {
+      description: {
+        story: generateStoryDescription('range', 'custom-track-colors'),
+      },
+    },
+  },
+  render: () => html`
+    <syn-range
+      class="custom-track-color"
+      max="100"
+      min="0"
+      value="50"
+    ></syn-range>
+    <style>
+      .custom-track-color {
+        --track-color-active: var(--syn-color-success-600);
+      }
+    </style>
+  `,
+};
+
+export const CustomTrackOffset: Story = {
+  parameters: {
+    docs: {
+      description: {
+        story: generateStoryDescription('range', 'custom-track-offset'),
+      },
+    },
+  },
+  render: () => html`
+    <syn-range
+      class="custom-track-offset"
+      max="100"
+      min="0"
+      value="25"
+    ></syn-range>
+    <style>
+      .custom-track-offset {
+        --track-active-offset: 50%;
+      }
+    </style>
+  `,
+};
+
+export const MultiKnob: Story = {
+  parameters: {
+    docs: {
+      description: {
+        story: generateStoryDescription('range', 'multi-knob'),
+      },
+    },
+  },
+  render: () => html`
+    <syn-range max="100" min="0" value="30 70"></syn-range>
+  `,
+};
+
+export const MultiKnobWithRestrictedMovement = {
+  parameters: {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    docs: {
+      description: {
+        story: generateStoryDescription('range', 'multi-knob-restrict-movement'),
+      },
+    },
+  },
+  render: () => html`
+    <syn-range
+      class="range-restrict-movement"
+      value="30 70"
+      label="Demo of restricting values"
+      min="0"
+      max="100"
+      step="1"
+    ></syn-range>
+    <script type="module">
+      Array
+        .from(document.querySelectorAll('.range-restrict-movement'))
+        .forEach((el) => {
+          el.addEventListener('syn-move', e => {
+            const { detail, target } = e;
+            const values = target.valueAsArray;
+            const { knob, value } = detail;
+
+            if (knob === target.knobs[0] && value > values[1]) {
+              e.preventDefault();
+            }
+
+            if (knob === target.knobs[1] && value < values[0]) {
+              e.preventDefault();
+            }
+          });
+        });
+    </script>
+  `,
+};
+
+export const Ticks: Story = {
```

Comment:
I just realized: I think the middle of the knob should define point 0, not the left side. Same for the tick. (That's how we defined in Figma.) But maybe we have to recheck. I'm aware that Shoelace and HTML does it differently.

![CleanShot 2024-07-16 at 10 32 07](https://github.com/user-attachments/assets/ac123e01-c49c-4c44-ae4d-0ee3f117c826)

Reply:
Its quite tricky to make this effect work in the first place. In this cse, you are suggesting to start the track a bit further on the right (on the first tick?). If this is the case, the widget itself would get smaller and not aligned with the label anymore, am I correct?

Reply:
No, the knob would overlap the label, which makes the tick align with the label as well.

Reply:
Get it, I will see what I can do :)